src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const storyOfActions = [];
+  const changeableObj = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(changeableObj, action.extraData);
+        storyOfActions.push({ ...changeableObj });
```

Comment:
```suggestion
```

Don't repeat yourself. Write `storyOfActions.push({ ...changeableObj });` only once - after `switch...case`. There will be only one operation executed, so you need only to push the result once